File: io/avro/src/test/resources/compat/issue-754/src/java/LegacyAvroReflectImplementation.java
package io.bluetape4k.avro.compat.issue754.java;

import io.bluetape4k.avro.AvroReflectSerializer;
import java.nio.charset.StandardCharsets;

public final class LegacyAvroReflectImplementation implements AvroReflectSerializer {
    public int serializeCalls;
    public int deserializeCalls;

    @Override
    public <T> byte[] serialize(T graph) {
        serializeCalls++;
        return graph == null ? null : "reflect".getBytes(StandardCharsets.UTF_8);
    }

    @Override
    public <T> T deserialize(byte[] bytes, Class<T> clazz) {
        deserializeCalls++;
        return bytes == null || bytes.length == 0 ? null : clazz.cast("reflect");
    }
}

File: io/avro/src/test/resources/compat/issue-754/src/kotlin/LegacyAvroReflectImplementation.kt
package io.bluetape4k.avro.compat.issue754.kotlin

import io.bluetape4k.avro.AvroReflectSerializer

class LegacyAvroReflectImplementation: AvroReflectSerializer {
    override fun <T> serialize(graph: T?): ByteArray? = graph?.toString()?.encodeToByteArray()

    @Suppress("UNCHECKED_CAST")
    override fun <T> deserialize(avroBytes: ByteArray?, clazz: Class<T>): T? =
        avroBytes?.takeIf { it.isNotEmpty() }?.decodeToString() as T?
}
